dali/pipeline/operator/op_schema.h

```diff
@@ -641,6 +655,34 @@ class DLL_PUBLIC OpSchema {
     return deprecated_in_favor_of_;
   }
 
+  DLL_PUBLIC inline bool IsDeprecatedArg(const std::string& arg_name) const  {
+    return deprecated_arguments_.find(arg_name) != deprecated_arguments_.end();
+  }
+
+  DLL_PUBLIC inline std::string DeprecatedArgMsg(const std::string& arg_name) const  {
+    auto it = deprecated_arguments_.find(arg_name);
+    DALI_ENFORCE(it != deprecated_arguments_.end(),
+      make_string("Argument \"", arg_name, "\" is not marked as a deprecated argument"));
+    if (!it->second.deprecated_str.empty())
+      return it->second.deprecated_str;
+    return make_string("Argument \"", arg_name, "\" for operator \"", name_,
+                       "\" is now deprecated.");
+  }
+
+  DLL_PUBLIC inline std::string DeprecatedArgRenamedTo(const std::string& arg_name) const  {
+    auto it = deprecated_arguments_.find(arg_name);
+    DALI_ENFORCE(it != deprecated_arguments_.end(),
+      make_string("Argument \"", arg_name, "\" is not marked as a deprecated argument"));
+    return it->second.renamed_to;
+  }
+
+  DLL_PUBLIC inline bool DeprecatedArgIgnore(const std::string& arg_name) const  {
+    auto it = deprecated_arguments_.find(arg_name);
+    DALI_ENFORCE(it != deprecated_arguments_.end(),
+      make_string("Argument \"", arg_name, "\" is not marked as a deprecated argument"));
+    return it->second.ignore;
+  }
```

Comment:
You can consider taking a look at what @JanuszL did recently with the oprator metadata and return this as a dictionary from one function call.